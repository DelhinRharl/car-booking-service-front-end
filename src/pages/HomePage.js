import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import ResponsiveCarousel from '../components/Carousel';
import { fetchCars } from '../redux/cars/carsSlice';

const HomePage = () => {
  const dispatch = useDispatch();
  const cars = useSelector((state) => state.cars.cars.length);
  useEffect(() => {
    dispatch(fetchCars());
  }, []);

  return (
    <section className="w-full md:w-4/5 md:py-20">
      <div className="w-[98%] text-center mb-10">
        <h1 className="font-bold text-2xl md:text-5xl md:mb-5">
          FEATURED MODELS
        </h1>
        <p className="font-bold text-sm text-gray-500">
          CHOOSE YOUR PREMIUM CAR.
        </p>
        <div>
          {cars}
          <span> cars</span>
        </div>
      </div>
      <ResponsiveCarousel />
    </section>
  );
};

export default HomePage;
