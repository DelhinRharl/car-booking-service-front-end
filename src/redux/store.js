import { configureStore } from '@reduxjs/toolkit';
import carsReducer from './cars/carsSlice';
import userReducer from './users/userSlice';

export default configureStore({
  reducer: {
    cars: carsReducer,
    user: userReducer,
  },
});
