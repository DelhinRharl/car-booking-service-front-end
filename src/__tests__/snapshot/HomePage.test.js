import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import HomePage from '../../pages/HomePage';
import { Provider } from 'react-redux';
import store from '../../redux/store';


describe('Home Page', () => {
  it('should render the  page correctly ', () => {
    const home = render(
        <Provider store={store}>
      <BrowserRouter>
        <HomePage />
      </BrowserRouter>,
    </Provider>
    );

    expect(home).toMatchSnapshot();
  });
});