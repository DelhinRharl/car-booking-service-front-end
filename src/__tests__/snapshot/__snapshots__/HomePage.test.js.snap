// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page should render the  page correctly  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="w-full md:w-4/5 pt-20 md:py-20 overflow-y-auto h-screen"
      >
        <div
          class="w-[98%] text-center mb-10"
        >
          <h1
            class="font-bold text-2xl md:text-5xl md:mb-5"
          >
            FEATURED MODELS
          </h1>
          <p
            class="font-bold text-sm text-gray-500"
          >
            CHOOSE YOUR PREMIUM CAR.
          </p>
        </div>
        <section
          class="py-10 md:hidden"
        />
        <section
          class="py-10 hidden md:block"
        />
      </section>
      ,
    </div>
  </body>,
  "container": <div>
    <section
      class="w-full md:w-4/5 pt-20 md:py-20 overflow-y-auto h-screen"
    >
      <div
        class="w-[98%] text-center mb-10"
      >
        <h1
          class="font-bold text-2xl md:text-5xl md:mb-5"
        >
          FEATURED MODELS
        </h1>
        <p
          class="font-bold text-sm text-gray-500"
        >
          CHOOSE YOUR PREMIUM CAR.
        </p>
      </div>
      <section
        class="py-10 md:hidden"
      />
      <section
        class="py-10 hidden md:block"
      />
    </section>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
